build/tools/release.sh

```diff
@@ -0,0 +1,129 @@
+#! /bin/sh
+set -e
+
+# this function will pint a help message
+help () {
+	echo "RELEASE script of kubeedge"
+	echo
+	echo "Usage:"
+	echo -e "\tthe first argument should be the version, which should be released"
+	echo -e "\tthe second argument is the path to the destination dir where the created files should be stored"
+}
+
+# this file will generate the checksum of an given file and store it into $filename.sum
+# this scirpt will use shasum to generate the shasum over the file
+# @param filename
+generateChecksum () {
+	file=$1
+	filename=$(echo checksum_$file.txt)
+	/usr/bin/shasum -a 512 $file  > $filename
+}
+
+# this function will move the builded file and the the checksum to the destination directory
+# @param destination directory
+# @param filename
+move () {
+	desDir=$2
+	filename=$1
+	checksum=$(echo checksum_$filename.txt)
+
+	mkdir -p $desDir
+	mv $filename $desDir/
+	mv $checksum $desDir/
+}
+
+# this function will generate the tar directories' sign the data and move the data to destDir
+# @param version
+# @param arch
+# @param desDir
+generateRel () {
+	version=$1
+	arch=$2
+	desDir=$3
+
+	keadm=$(echo keadm-$version-linux-$arch)
+	edge=$(echo edge-$version-linux-$arch)
+	kubeedge=$(echo kubeedge-$version-linux-$arch)
+	edgesite=$(echo edgesite-$version-linux-$arch)
+
+	cd $PWD 
+
+	mkdir -p $edge/edge
+	echo $version > $edge/version
+	mkdir -p $kubeedge/edge
+	echo $version > $kubeedge/version
+	mkdir -p $kubeedge/cloud
+	mkdir -p $kubeedge/keadm
+	mkdir $keadm
+	echo $version > $keadm/version
+	mkdir -p $edgesite
+	echo $version > $edgesite/version
+
+	# copy edge core to the expected directories
+	echo copy edge
+	cp edge/edge_core $edge/edge/
+	cp -r edge/conf $edge/edge/
+	cp edge/edge_core $kubeedge/edge/
+	cp -r edge/conf $kubeedge/edge/
+
+	# copy edgecontroller to the expected directories
+	cp cloud/edgecontroller $kubeedge/cloud/
+	cp -r cloud/conf $kubeedge/cloud/
+
+	# copy keadm to the expected directories
+	cp keadm/keadm $keadm/
+
+	# copy to edgesite
+	cp edgesite/edgesite $edgesite/
+	cp -r edgesite/conf $edgesite/
+
+	tar cfv $(echo $keadm.tar) $keadm
```

Comment:
can we compress further and make all of them .tar.gz ?

Reply:
Yes of course